kubernetes/samples/scripts/common/wdt-utility.sh

```diff
@@ -0,0 +1,389 @@
+#!/bin/bash
+# Copyright (c) 2021, Oracle and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+#
+# Description:
+#
+#   This script contains functions for installing WebLogic Deploy Tool (WDT) and
+#   WebLogic Image Tool (WIT), and for running WDT.
+#
+#
+# Usage:
+#
+#   Export customized values for the input shell environment variables as needed
+#   before calling this script.   
+#
+# Outputs:
+#
+#   WDT install:           WDT_DIR/weblogic-deploy/...
+#
+#   Copy of wdt model:     WDT_DIR/$(basename WDT_MODEL_FILE)
+#   Copy of wdt vars:      WDT_DIR/$(basename WDT_VAR_FILE)
+#
+#   WDT logs:              WDT_DIR/weblogic-deploy/logs/...
+#   WDT stdout:            WDT_DIR/createDomain.sh.out
+#
+#   WebLogic domain home:  DOMAIN_HOME_DIR
+#                          default: /shared/domains/<domainUID>
+#
+# Input environment variables:
+#
+#   ORACLE_HOME    Oracle home with a WebLogic install.
+#                  default:  /u01/oracle
+#
+#   DOMAIN_HOME_DIR  Target location for generated domain.
+#
+#   WDT_MODEL_FILE Full path to WDT model file.
+#                  default:  the directory that contains this script
+#                            plus "/wdt_model.yaml"
+#
+#   WDT_VAR_FILE   Full path to WDT variable file (java properties format).
+#                  default:  the directory that contains this script
+#                            plus "/create-domain-inputs.yaml"
+#
+#   WDT_DIR        Target location to install and run WDT, and to keep a copy of
+#                  $WDT_MODEL_FILE and $WDT_MODEL_VARS. Also the location
+#                  of WDT log files.
+#                  default:  /shared/wdt
+#
+#   WDT_VERSION    WDT version to download.
```

Comment:
(1) Maybe include support for 'latest' for both WIT_VERSION and WDT_VERSION?
(2) Also, perhaps 'latest' should be the default?

In  which case the URLs become:
```
https://github.com/oracle/weblogic-deploy-tooling/releases/latest/download/weblogic-deploy.zip
https://github.com/oracle/weblogic-image-tool/releases/latest/download/imagetool.zip
```

Reply:
Oh scratch that. I see this is an old script that already supports latest, but in a different way.  Please document this script's existing support for the "LATEST" keyword here, and also consider simplifying script to use the URLs above instead of attempting to parse out the latest version from a curl to 'releases'.

Reply:
I was refering to the version for curl download - not the cache entry when I strted this thread.  As for the cache entry, I think it used to be it was necessary to delete old key before replacing it, but I think WDT now allows this, also, the WIT command itself implicitly defaults to 'wdt_latest'.